File: src/main/java/com/vjiki/music/dto/SongResponse.java
package com.vjiki.music.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SongResponse {
    private String id;
    private String artist;
    
    @JsonProperty("audio_url")
    private String audioUrl;
    
    private String cover;
    private String title;
}

File: src/main/kotlin/com/vjiki/music/dto/SongResponse.kt
package com.vjiki.music.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class SongResponse(
    val id: String,
    val artist: String?,
    @JsonProperty("audio_url")
    val audioUrl: String?,
    val cover: String?,
    val title: String?,
    val isLiked: Boolean = false,
    val isDisliked: Boolean = false,
    val likesCount: Long = 0L,
    val dislikesCount: Long = 0L
)

